bundles/org.openhab.binding.pushbullet/src/main/java/org/openhab/binding/pushbullet/internal/handler/PushbulletHandler.java

```diff
@@ -0,0 +1,218 @@
+        logger.debug("CFG {}", configuration);
+        logger.debug("CFG Name  '{}'", configuration.getName());
+
+        try {
+            logger.debug("Recipient is '{}'", recipient);
+            logger.debug("Title is     '{}'", title);
+            logger.debug("Message is   '{}'", message);
+
+            Push push = new Push();
+            push.setTitle(title);
+            push.setBody(message);
+            push.setType(type);
+
+            if (recipient != null) {
+                if (isValidEmail(recipient)) {
+                    logger.debug("Recipient is an email address");
+                    push.setEmail(recipient);
+                } else if (isValidChannel(recipient)) {
+                    logger.debug("Recipient is a channel tag");
+                    push.setChannel(recipient);
+                } else {
+                    logger.warn("Invalid recipient: {}", recipient);
+                    logger.warn("Message will be broadcast to all user's devices.");
+                }
+            }
+
+            logger.debug("Push: {}", push);
+
+            String request = gson.toJson(push);
+            logger.debug("Packed Request: {}", request);
+
+            Properties headers = new Properties();
+            headers.put(HttpHeader.USER_AGENT, "openHAB / Pushbullet binding " + VERSION.toString());
+            headers.put(HttpHeader.CONTENT_TYPE, MimeTypes.Type.APPLICATION_JSON.asString());
+            headers.put("Access-Token", configuration.getToken());
+
+            logger.debug("Headers: {}", headers);
+
+            try (InputStream stream = new ByteArrayInputStream(request.getBytes(StandardCharsets.UTF_8))) {
+
+                String pushAPI = configuration.getApiUrlBase() + "/" + API_METHOD_PUSHES;
+
+                String responseString = HttpUtil.executeUrl(HttpMethod.POST.asString(), pushAPI, headers, stream,
+                        MimeTypes.Type.APPLICATION_JSON.asString(), TIMEOUT);
+
+                logger.debug("Got Response: {}", responseString);
+                PushResponse response = gson.fromJson(responseString, PushResponse.class);
+
+                logger.debug("Unpacked Response: {}", response);
+
+                stream.close();
+
+                if ((null != response) && (null == response.getPushError())) {
+                    result = true;
+                }
+            }
+            catch (IOException e) {
+                logger.warn("IO problems pushing note: {}", e.getMessage());
+            }
+
+        } catch (java.lang.Error e) {
+            if (e.getCause() instanceof HttpResponseException) {
+                logger.debug("Possibly expired token, check on web site", e);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "@text/offline.conf-error-httpresponseexception");
+            }
+
+            logger.debug("Error in getResponseString: ", e);
+        }
+
+        return result;
```

Comment:
Consider splitting up this method

Reply:
But how? logically, it is one operation. splitting it up would only serve to introduce multiple methods, but to no gain with regard to the logic that has to be performed.

Reply:
I would say that it contains a lot of logic units, eg contructing the push message, or constructing and convert it to a string. Or performing the actual request.

Reply:
I'll refactor it later tonight, at the same time working on the error-catching part.